File: IOS Code coversion/TOCColors.swift
import SwiftUI

// Dictionary mapping TOC names to their HEX color codes
let tocColors: [String: String] = [
    "Standard": "#868686",
    "First": "#9800F0",
    "Greater Anglia": "#d70428",
    "ScotRail": "#1e467d",
    "Avanti West Coast": "#004354",
    "c2c": "#b7007c",
    "Caledonian Sleeper": "#1d2e35",
    "Chiltern Railways": "#00bfff",
    "CrossCountry": "#660f21",
    "East Midlands Railway": "#713563",
    "Great Western Railway": "#0a493e",
    "Hull Trains": "#de005c",
    "Thameslink/Great Northern": "#ff5aa4",
    "Heathrow Express": "#532e63",
    "Island Line": "#1e90ff",
    "Transport for Wales": "#ff0000",
    "LNER": "#ce0e2d",
    "Northern": "#034DE2",
    "TransPennine Express": "#09a4ec",
    "Multi-Operator": "#F1F1F1",
    "Merseyrail": "#fff200",
    "Gatwick Express": "#EB1E2D",
    "Great Northern": "#ff5aa4",
    "LNWR": "#00bf6f",
    "South Western Railway": "#24398c",
    "Southeastern": "#389cff",
    "Southern": "#8cc63e",
    "Thameslink": "#ff5aa4",
    "West Midlands Trains": "#ff8300",
    "Lumo": "#2b6ef5",
    "Grand Central": "#1d1d1b",
    "London Overground": "#e87722",
    "London Underground": "#10069f",
    "Elizabeth Line": "#6950a1",
    "Bee Network": "#FFCC33",
    "Blackpool Tramway": "#7F2680",
    "Docklands Light Railway (DLR)": "#00b2a9",
    "Edinburgh Trams": "#8D122A",
    "Glasgow Subway": "#f57c14",
    "London Tramlink": "#78be20",
    "Manchester Metrolink": "#edb600",
    "Nottingham Express Transit (NET)": "#01796f",
    "Sheffield Supertram": "#C67A1E",
    "Tyne & Wear Metro": "#FBB10F",
    "West Midlands Metro": "#0075c9",
    "Heritage": "#800000",
    "International": "#355e3b",
    "Eurostar": "#086bfe",
    // Add more TOCs and their colors as needed
]

// Extension to initialize Color with a HEX string
extension Color {
    init?(hex: String) {
        var hexSanitized = hex.trimmingCharacters(in: .whitespacesAndNewlines)
        hexSanitized = hexSanitized.replacingOccurrences(of: "#", with: "")

        var rgb: UInt64 = 0
        guard Scanner(string: hexSanitized).scanHexInt64(&rgb) else { return nil }

        let red = Double((rgb >> 16) & 0xFF) / 255.0
        let green = Double((rgb >> 8) & 0xFF) / 255.0
        let blue = Double(rgb & 0xFF) / 255.0

        self.init(red: red, green: green, blue: blue)
    }
}

// Function to get a Color for a TOC
func colorForTOC(_ toc: String?) -> Color? {
    guard let toc = toc, let hex = tocColors[toc] else {
        return nil // Return nil if TOC or HEX code is not found
    }
    return Color(hex: hex)
}

File: app/src/main/java/com/jw/railstatistics/data/TOCColors.kt
package com.jw.railstatistics.data

import android.graphics.Color

object TOCColors {
    // Dictionary mapping TOC names to their HEX color codes
    private val tocColors: Map<String, String> = mapOf(
        "Standard" to "#868686",
        "First" to "#9800F0",
        "Greater Anglia" to "#d70428",
        "ScotRail" to "#1e467d",
        "Avanti West Coast" to "#004354",
        "c2c" to "#b7007c",
        "Caledonian Sleeper" to "#1d2e35",
        "Chiltern Railways" to "#00bfff",
        "CrossCountry" to "#660f21",
        "East Midlands Railway" to "#713563",
        "Great Western Railway" to "#0a493e",
        "Hull Trains" to "#de005c",
        "Thameslink/Great Northern" to "#ff5aa4",
        "Heathrow Express" to "#532e63",
        "Island Line" to "#1e90ff",
        "Transport for Wales" to "#ff0000",
        "LNER" to "#ce0e2d",
        "Northern" to "#034DE2",
        "TransPennine Express" to "#09a4ec",
        "Multi-Operator" to "#F1F1F1",
        "Merseyrail" to "#fff200",
        "Gatwick Express" to "#EB1E2D",
        "Great Northern" to "#ff5aa4",
        "LNWR" to "#00bf6f",
        "South Western Railway" to "#24398c",
        "Southeastern" to "#389cff",
        "Southern" to "#8cc63e",
        "Thameslink" to "#ff5aa4",
        "West Midlands Trains" to "#ff8300",
        "Lumo" to "#2b6ef5",
        "Grand Central" to "#1d1d1b",
        "London Overground" to "#e87722",
        "London Underground" to "#10069f",
        "Elizabeth Line" to "#6950a1",
        "Bee Network" to "#FFCC33",
        "Blackpool Tramway" to "#7F2680",
        "Docklands Light Railway (DLR)" to "#00b2a9",
        "Edinburgh Trams" to "#8D122A",
        "Glasgow Subway" to "#f57c14",
        "London Tramlink" to "#78be20",
        "Manchester Metrolink" to "#edb600",
        "Nottingham Express Transit (NET)" to "#01796f",
        "Sheffield Supertram" to "#C67A1E",
        "Tyne & Wear Metro" to "#FBB10F",
        "West Midlands Metro" to "#0075c9",
        "Heritage" to "#800000",
        "International" to "#355e3b",
        "Eurostar" to "#086bfe"
    )

    // Function to get a Color for a TOC
    fun getColorForTOC(toc: String?): Int? {
        if (toc == null) return null
        val hex = tocColors[toc] ?: return null
        return try {
            Color.parseColor(hex)
        } catch (e: IllegalArgumentException) {
            null
        }
    }

    // Function to get all TOC names
    fun getAllTOCNames(): List<String> {
        return tocColors.keys.toList().sorted()
    }
} 